File: frontend/src/types/exhibition.ts
import { ExhibitionStatus, ExhibitionType } from './enums';

export interface Exhibition {
  id: string;
  title: string;
  description: string;
  curatorId: string;
  startDate: string;
  endDate: string;
  type: ExhibitionType;
  coverUrl: string;
  artworkIds: string[];
  status: ExhibitionStatus;
  visitors: number;
}

export interface ExhibitionReadiness {
  exhibitionId: string;
  totalArtworks: number;
  publicReadyCount: number;
  pendingReviewCount: number;
  rejectedCount: number;
  draftCount: number;
  soldCount: number;
  archivedCount: number;
  noImageCount: number;
  missingCount: number;
  readyRatio: number;
  ready: boolean;
  blockingReasons: string[];
}

File: backend/internal/model/exhibition.go
package model

import "time"

// Exhibition 展览。
type Exhibition struct {
	ID           string    `bson:"_id" json:"id"`
	Title        string    `bson:"title" json:"title"`
	Description  string    `bson:"description" json:"description"`
	CuratorID    string    `bson:"curator_id" json:"curatorId"`
	StartDate    string    `bson:"start_date" json:"startDate"`
	EndDate      string    `bson:"end_date" json:"endDate"`
	Type         string    `bson:"type" json:"type"`
	CoverURL     string    `bson:"cover_url" json:"coverUrl"`
	ArtworkIDs   []string  `bson:"artwork_ids" json:"artworkIds"`
	Status       string    `bson:"status" json:"status"`
	Visitors     int64     `bson:"visitors" json:"visitors"`
	ReviewStatus string    `bson:"review_status" json:"reviewStatus"`
	CreatedAt    time.Time `bson:"created_at" json:"createdAt"`
	UpdatedAt    time.Time `bson:"updated_at" json:"updatedAt"`
}
